tests/suites/test_suite_psa_crypto.function

```diff
@@ -8954,6 +8954,25 @@ void derive_input(int alg_arg,
     }
     TEST_EQUAL(actual_output_status, expected_output_status);
 
+    /* Test calling input functions after operation has been aborted
```

Comment:
Thanks for adding a test! But please don't add extra code to this function, which is already more complex than I'd like.

Proposal 1: keep this function, but skip the call to `psa_key_derivation_setup()` when `alg == 0`. Add test cases that expect that each input attempt returns `BAD_STATE`.

Proposal 2: make a new function that tries `input_xxx` without `setup`.